src/Ouroboros/Network/MonadClass/MonadSTM.hs

```diff
@@ -87,6 +91,74 @@ class (MonadFork m, Monad (Tr m)) => MonadSTM m where
   lengthTBQueue  :: TBQueue m a -> Tr m Natural
 
 
+instance MonadFork m => MonadFork (ReaderT e m) where
+  fork (ReaderT f) = ReaderT $ \e -> fork (f e)
+
+instance (MonadFork m, MonadSTM m) => MonadSTM (ReaderT e m) where
```

Comment:
Do we need `MonadFork m` here, `MonadSTM m` should already bring it, isn't it?

Reply:
Ha, good catch! I will clean this up.